llama-index-core/llama_index/core/workflow/service.py

```diff
@@ -16,12 +16,16 @@ class ServiceManager:
     The service is made available to the steps of the main workflow.
     """
 
-    _services: Dict[str, "Workflow"] = {}
+    def __init__(self) -> None:
+        self._services: Dict[str, "Workflow"] = {}
 
-    def get(self, name: str) -> "Workflow":
+    def get(self, name: str, default: Optional["Workflow"] = None) -> "Workflow":
         try:
             return self._services[name]
         except KeyError as e:
+            if default:
+                return default
```

Comment:
should we register `default` in `self._services[name]` now?

Reply:
We might but if `default` appears in more than one step, those instances would clash. If we forget about the default value, each step can have its own default, but if we decide to store it, we should validate that every occurrence of the same "nested workflow" name across all the steps have the same default value.

Reply:
ah okay, thanks for the explanation! I'm good with either route as I lack enough context here to have an opinion on whether storing `default` is necessary at this point. Maybe we forget for now and decide to store it if we learn that we need to?